### PR TITLE
Remove KPrint object from KCFGExplore

### PR DESCRIPTION
**Pull request by ehildenb**
Currently, the `KCFGExplore` object holds both a `KPrint` and `CTermSymbolic`. But it really only needs `KPrint` for the `pretty_print` functionality, which only really relies on the `PrettyPrinter` class in the `kast/pretty` module. So this PR reduces that dependency a bit.

- Places where `KCFGExplore` needed a `KDefinition` now reuse the one held by `CTermSymbolic`.
- `KCFGExplore._pretty_printer` is added for building a `PrettyPrinter` directly. `KCFGExplore.pretty_print` is added to expose the pretty printing functionality of that `PrettyPrinter`.
- Tests and other files are updated.

### CODE REVIEW

File: src/pyk/testing/_kompiler.py [hunk 1]
```diff
@@ -253,11 +253,11 @@ def semantics(self, definition: KDefinition) -> KCFGSemantics:
     def kcfg_explore(
         self,
         cterm_symbolic: CTermSymbolic,
-        kprint: KProve,
+        kprint: KPrint,
```
Comment (tothtamas28) [inline src/pyk/testing/_kompiler.py:256]: This no longer needs to be a `KPrintTest`.

File: src/pyk/kcfg/explore.py [hunk 2]
```diff
@@ -40,25 +41,29 @@
...
         id: str | None = None,
     ):
-        self.kprint = kprint
         self.cterm_symbolic = cterm_symbolic
         self.kcfg_semantics = kcfg_semantics if kcfg_semantics is not None else DefaultSemantics()
         self.id = id if id is not None else 'NO ID'
 
+    @cached_property
+    def _pretty_printer(self) -> PrettyPrinter:
+        return PrettyPrinter(self.cterm_symbolic._definition)
```
Comment (tothtamas28) [inline src/pyk/kcfg/explore.py:62]: Does this handle the case when we need to customize the symbol table?

Comment (ehildenb) [reply src/pyk/kcfg/explore.py:62]: No. But the pretty printer in the `KCFGExplore` is just being used for printing out some log and error messages. Specifically:

- When building `implication_failure_reason`, it's used to print out what is almost 100% going to be boolean predicates (so likely default printer will work fine).
- When getting branches during KCFG exploration, it's used to log the branch conditions (boolean conditions).

I think long term, all printing functionality should be removed from this class and done at a higher level.

File: src/pyk/testing/_kompiler.py (hunk 1 above)
Comment (ehildenb) [reply src/pyk/testing/_kompiler.py]: Done.

**Review by tothtamas28 — APPROVED** [review]

**Comment by tothtamas28** [comment]
Please do not merge until runtimeverification/evm-semantics#2302 goes through.

**Pull request merged**